File: clients/ios/MPTD/MPTD/Economy.m
//
//  Economy.m
//  MPTD
//
//  Created by Karl-Johan Alm on 4/26/12.
//  Copyright (c) 2012 __MyCompanyName__. All rights reserved.
//
#if 0

#import "Economy.h"
#import "Monster.h"
#import "Player.h"

@implementation Economy

@synthesize mpIncome = _mpIncome;
@synthesize me = _me;

- (id)init
{
    self = [super init];
    if (self) {
        _spIncome = 10;
        _mpIncome = 10;
    }
    return self;
}

- (int)getSPIncome:(int)waveCount
{
    _spIncome += waveCount - 1;
    _me.income = _spIncome;
    return _spIncome;
}

- (int)getMonsterMoney:(Monster *)mon
{
    return pow((mon.maxHP + 32.f * mon.speed) / 54.f, 0.8f);
}

- (int)buyMonster:(Monster *)mon
{
    _mpIncome += pow(((double)mon.maxHP + (mon.maxHP / 3 * mon.speed)) / 104, 0.8);
    _me.income = _mpIncome;
    return _mpIncome;
}

@end
#endif

File: clients/windesktop/C-SharpClient_1.1/C-SharpClient_1.1/Economy.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace C_SharpClient_1._1
{
    class Economy
    {
        //income multi and single player is increased later on by their respective function 10 is the amount of money recieved after first round + little extra
        public int sPIncome = 10;
        public int mPIncome = 10;
        //extra amount of money players income is increased for buying monsters in multiplayer
        // kanske ha senare private int mPMonsterMoney;

        public int getSPIncome(int rounds)
        {
            return sPIncome += rounds -1;
        }

        /// <summary>
        /// Get the ammount of money you get from one monster!
        /// </summary>
        /// <param name="mon">The monster you want to generate money from</param>
        /// <returns></returns>
        public int getMonsterMoney(Monster mon)
        {
            return Convert.ToInt32(Math.Pow(((double)mon.maxHP + (mon.maxHP / 3) * mon.maxSpeed) / 104, 0.8) * 3 - 2);
        }

        /// <summary>
        /// Increase the income based on what monster was bought 
        /// going to fix the buy function
        /// </summary>
        /// <param name="mon">the monster that was bought</param>
        /// <returns>the increased ammount of money</returns>
        public int BuyMonster(Monster mon)
        {
            return mPIncome += Convert.ToInt32(Math.Pow(((double)mon.maxHP + (mon.maxHP / 3 * mon.maxSpeed)) / 104, 0.8));
        }

        // return the income to a player done in begining or end of each round
        public int GetMultiPlayerIncome()
        {
            return mPIncome;
        }
    }
}
